src/Psalm/Issue/MixedAssignment.php

```diff
@@ -1,11 +0,0 @@
-<?php
-
-namespace Psalm\Issue;
-
-final class MixedAssignment extends CodeIssue implements MixedIssue
```

Comment:
The class has to remain here, but be marked as `@deprecated`. It's part of our API surface.

Reply:
I think it should be safe to remove the class between major versions — I don't expect plugins to rely on its existence.

Reply:
I can confirm none of the plugins available on Packagist use `MixedAssignment` in the code:
```console
> fetch https://packagist.org/packages/list.json?type=psalm-plugin | get packageNames | each {|it| fetch $"https://packagist.org/packages/($it).json"} | get package.repository | each { str replace 'https://github.com/' ''} | save plugin_repos.json

> open plugin_repos.json | each {|it| {repo: $it dir: ($it | str replace "/" "---")} } | each {|it| gh repo clone $it.repo $it.dir }

> rg --pcre2 '(?<!-suppress )(?<!\<)MixedAssignment'
psalm---psalm-plugin-symfony/tests/acceptance/acceptance/Response.feature
7:  Scenario: MixedAssignment error about $statusTexts is not raised

psalm---psalm-plugin-symfony/tests/acceptance/acceptance/DenormalizerInterface.feature
41:      | MixedAssignment        | Unable to determine the type that $result is being assigned to |

psalm---psalm-plugin-symfony/tests/acceptance/acceptance/serializer/SerializerInterface6.feature
41:      | MixedAssignment        | Unable to determine the type that $result is being assigned to |

psalm---psalm-plugin-symfony/tests/acceptance/acceptance/console/ConsoleOption.feature
297:      | MixedAssignment | Unable to determine the type that $option is being assigned to |

psalm---psalm-plugin-symfony/tests/acceptance/acceptance/serializer/SerializerInterface.feature
41:      | MixedAssignment        | Unable to determine the type that $result is being assigned to |

psalm---psalm-plugin-symfony/tests/acceptance/acceptance/forms/FormGetErrors.feature
70:      | MixedAssignment         | Unable to determine the type of this assignment                                     |

psalm---symfony-psalm-plugin/tests/acceptance/acceptance/console/ConsoleOption.feature
297:      | MixedAssignment | Unable to determine the type that $option is being assigned to |

psalm---symfony-psalm-plugin/tests/acceptance/acceptance/forms/FormGetErrors.feature
70:      | MixedAssignment         | Unable to determine the type of this assignment                                     |

psalm---symfony-psalm-plugin/tests/acceptance/acceptance/DenormalizerInterface.feature
41:      | MixedAssignment        | Unable to determine the type that $result is being assigned to |

psalm---symfony-psalm-plugin/tests/acceptance/acceptance/Response.feature
7:  Scenario: MixedAssignment error about $statusTexts is not raised

psalm---symfony-psalm-plugin/tests/acceptance/acceptance/serializer/SerializerInterface6.feature
41:      | MixedAssignment        | Unable to determine the type that $result is being assigned to |

psalm---symfony-psalm-plugin/tests/acceptance/acceptance/serializer/SerializerInterface.feature
41:      | MixedAssignment        | Unable to determine the type that $result is being assigned to |

Practically---PsalmPluginYii2/tests/acceptance/PropertyAccessor.feature
136:      | MixedAssignment | Unable to determine the type that $contentType is being assigned to |
```

Two use it in their tests, as shown above.